### PR TITLE
Adding reset Gid to default when local volume gets unmounted

### LINKED ISSUE #50024: local volume Gid does not get reset after pod termination
k8s version v1.8.0-alpha.2.1149+8dc7cf1ebd4515-dirty

When POD which was using a local volume terminates and this POD was using a non-default Gid, after POD termination Gid of volume does not reset to default. It causes a problem when another POD tries to mount this local volume as volume still has stale Gid.
I would like to propose to reset Gid to 0 on unmount operation of a local volume.

/sig storage

### PR DESCRIPTION
**Pull request by sbezverk**
Closes #50024

### CODE REVIEW

**Comment by msau42** [comment]
Should it be reset after every unmount, or only when the volume is destroyed?  What happens if a second pod is also using the volume at the same time the first one unmounted?

**Comment by sbezverk** [comment]
@msau42 Good point but since not always a volume gets delete, I would go with checking if other mount for this volume exist and reset gid only on last unmount. What do you think?

**Comment by msau42** [comment]
It's a user error if they have two pods that are configured with different fsgroup sharing the same volume.  So I think the best behavior we can have is that the first pod can set the fsgroup, and any other pod after that has to match.  Then the fsgroup can be reset once the volume is destroyed.

**Comment by msau42** [comment]
Checking for other pods mounting is also prone to race conditions, since mounts can happen in parallel.

**Comment by sbezverk** [comment]
@msau42 you assume that volume gets always destroyed, which I think is not always the case. In my testbed :) it never gets destroyed that is how I discovered it. I think even volume destroy will not help here because gid also gets assigned automatically to the mount point, so if you delete volume and then use the same mount point I am positive it will get old gid.

**Comment by msau42** [comment]
I think the fsgroup should not change once the volume is used by someone.  In your testbed, the volume, from a Kubernetes point of view, is destroyed.  The PV object is destroyed when you destroy your test cluster.

There's also the scenario where the admin created the volume with specific gid.  We do not want to reset it to a gid that the admin did not want.

File: pkg/volume/local/local.go [hunk 1]
```diff
@@ -229,6 +229,13 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 			return err
 		}
 		if !notMnt {
+			// Reset Gid of a local volume which will be unmounted next
+			if fsGroup != nil {
+				*fsGroup = 0
```
Comment (brendandburns) [inline pkg/volume/local/local.go]: nit: why assign to fsGroup here? why not just use inline `0` below

**Comment by msau42** [comment]
This PR should be closed and not merged.  It is not the behavior we want.

**Pull request closed without merging**